CMakeLists.txt

```diff
@@ -408,6 +409,59 @@ target_link_directories(notcurses PRIVATE ${OIIO_LIBRARY_DIRS})
 target_link_directories(notcurses-static PRIVATE ${OIIO_STATIC_LIBRARY_DIRS})
 endif()
 
+#######################################
+# libnotcurses-ffi (ffi shared library)
+file(GLOB NCFFISRCS CONFIGURE_DEPENDS src/libffi/*.c src/libffi/*.cpp)
+add_library(notcurses-ffi SHARED ${NCFFISRCS})
+target_compile_options(notcurses-ffi PUBLIC -fkeep-inline-functions)
+target_compile_definitions(notcurses-ffi PUBLIC NOTCURSES_FFI)
+# don't want these on freebsd/dragonfly/osx
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+target_compile_definitions(notcurses-ffi
```

Comment:
like i said, it would be nice to factor these out into some common definition, but you don't need to worry about that.